keanu-project/src/main/java/io/improbable/keanu/vertices/dbl/DoubleVertexSamples.java

```diff
@@ -39,4 +39,7 @@ public DoubleTensor getVariances() {
             .divInPlace(samples.size()-1);
     }
 
+    public DoubleTensor asTensor() {
+        return DoubleTensor.concat(samples.toArray(new DoubleTensor[0]));
```

Comment:
Very close but I don't think this behaviour is quite right. In your test we are doing an `asTensor` on 4 samples of shape `1x3`. This should return a `4x1x3` but instead we're getting a `4x3`. Sounds like concat is not adding on an extra dimension? You may have to do a reshape after the concat.

Reply:
Done.